### DISCUSSION TITLE
useFieldArray maintains defaultValues after fields removed

### DISCUSSION THREAD

**Discussion by alexw23** [react-hook-form/react-hook-form#5167] [Q&A] [answered]
The intention is to use `defaultValues` to set initial values of the fieldArray. At which point a user can add/remove fields as they like. The expectation is that after calling `remove(index)`, the next time a new item is added on that index it will not contain the defaultValue.

Sandbox is here:
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-5uebl?file=/src/index.js

On load all defaultValues exist. Remove row "Emma". Then add a row and, "Emma" reappears.

Expected Behaviour:
On load all defaultValues exist. Remove row "Emma". Then add a row and input is empty.

**Comment by bluebill1049** [comment]
<img width="950" alt="Screen Shot 2021-05-12 at 10 27 15 am" src="https://user-images.githubusercontent.com/10513364/117900497-9ffbf080-b30c-11eb-82c8-7247e8dfbacc.png">

have a read this section @alexw23 

>The field.id (and not index) must be added as the component key to prevent re-renders breaking the fields:

**Reply by alexw23** [reply]
Thanks @bluebill1049, I've updated the sandbox to match the above (field.id not index) and I'm still seeing the same issue.

**Comment by bluebill1049** [comment, marked as answer]
please send sometimes with the doc, append arg is required.

<img width="979" alt="Screen Shot 2021-05-12 at 10 42 47 am" src="https://user-images.githubusercontent.com/10513364/117901502-d0448e80-b30e-11eb-9f88-6c6efd253e32.png">

**Reply by alexw23** [reply]
Okay makes sense now thank you. Solution was to provide an empty object i.e. `append({firstName: null})`.
